### PR TITLE
[11.0] [MIG] date range with hierarchical feature

### PR DESCRIPTION
**Pull request by ntsirintanis**
Migrating hierarchical feature for date_range to Odoo 11.0.

### CODE REVIEW

File: date_range/__manifest__.py [hunk 1]
```diff
@@ -1,9 +1,9 @@
-# © 2016 ACSONE SA/NV (<http://acsone.eu>)
-# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+# © 2016 ACSONE SA/NV (<https://acsone.eu>)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
 {
     "name": "Date Range",
     "summary": "Manage all kind of date range",
-    "version": "11.0.2.0.1",
+    "version": "11.0.3.0.1",
```
Comment (lmignon) [inline date_range/__manifest__.py]: @ntsirintanis Can you also update the history as in https://github.com/OCA/server-tools/commit/ed05e386ac237f5b8f0344e5f6e7088609aae639#diff-3db8a615cdc543d077b5b3cfbc7f939d

**Review by lmignon — CHANGES_REQUESTED** [review]
Thank you @ntsirintanis for the forward port. 
Can you also update the history file since you change the addon version?

**Comment by ntsirintanis** [comment]
This is now ready for review. @lmignon @NL66278

File: date_range/__init__.py [hunk 2]
```diff
@@ -1,5 +1,5 @@
-# © 2016 ACSONE SA/NV (<http://acsone.eu>)
-# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+# © 2016 ACSONE SA/NV (<https://acsone.eu>)
```
Comment (NL66278) [inline date_range/__init__.py]: Copyright statements are to be removed from __init__.py files altogether.

File: date_range/__manifest__.py [hunk 3]
```diff
@@ -1,9 +1,9 @@
-# © 2016 ACSONE SA/NV (<http://acsone.eu>)
-# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+# © 2016 ACSONE SA/NV (<https://acsone.eu>)
```
Comment (NL66278) [inline date_range/__manifest__.py]: In remaining copyright lines, replace the '©' by the word 'Copyright'

File: date_range/models/date_range.py [hunk 4]
```diff
@@ -52,6 +58,45 @@ def _check_company_id_type_id(self):
                     _('The Company in the Date Range and in '
                       'Date Range Type must be the same.'))
 
+    @api.constrains('parent_id', 'date_start', 'date_end')
+    def _validate_child_range(self):
+        for this in self:
+            if not this.parent_id:
+                continue
+            start = this.parent_id.date_start <= this.date_start
```
Comment (NL66278) [inline date_range/models/date_range.py]: You can make your code much more readable by renaming 'start' to date_start_valid end 'end' to date_end_valid. Then, if date_start_valid and date_end_valid, just continue and you save an identation level. And you do not then need the strangely named child_range variable.

**Review by NL66278 — CHANGES_REQUESTED** [review]
See other comments

**Comment by lmignon** [comment]
@ntsirintanis Thank you again. Can you rebase your branch and resolve conflicts. An other one updating the headers has been merge a little bit too fast and your PR is now in conflict. Sorry for this additional work..

File: date_range/__manifest__.py [hunk 5]
```diff
@@ -4,6 +4,7 @@
     "name": "Date Range",
     "summary": "Manage all kind of date range",
     "version": "11.0.3.0.0",
+    "version": "11.0.2.0.1",
```
Comment (NL66278) [inline date_range/__manifest__.py]: Either do not change the version, or increment it.

**Review by NL66278 — APPROVED** [review]
:+1: LGTM Codereview. Tested previous version.

**Comment by NL66278** [comment]
@lmignon The conflicts have been resolved. Could you refresh your review and see wether anything left to be done?

**Comment by gfcapalbo** [comment]
LGTM

**Comment by NL66278** [comment]
@gfcapalbo If you wanted to review, you should use the Review button on the Files changed tab.

File: date_range/readme/HISTORY.rst [hunk 6]
```diff
@@ -0,0 +1,23 @@
+11.0.3.0.0 (2019-01-14)
```
Comment (kmatthes) [inline date_range/readme/HISTORY.rst]: To match the version number in the manifest
```suggestion
11.0.3.0.1 (2019-01-14)
```

**Review by kmatthes — COMMENTED** [review]
Code looks almost fine. I did a short test in runbot.

**Review by kmatthes — APPROVED** [review]
For me it's fine now.

**Review by kmatthes — CHANGES_REQUESTED** [review]
@ntsirintanis As far as I know the process, a migration should not include any improvements. For further improvements there should be a seperated PR.

**Comment by ntsirintanis** [comment]
@kmatthes this is the separate PR.

**Comment by kmatthes** [comment]
@ntsirintanis Sorry, I don't get it. This PR (#47) is for migrating a feature (from 10.0), isn't it?

**Comment by ntsirintanis** [comment]
@kmatthes see [here](https://github.com/OCA/server-ux/tree/11.0/date_range)

**Comment by kmatthes** [comment]
@ntsirintanis Yeah, I know. But your even advertised the change as a migration. **[MIG]** Then I guess we should follow the migration process. That mean since the last commit is not included in 10.0 yet, it's an improvement/change and not just a migration.

File: date_range/__manifest__.py [hunk 7]
```diff
@@ -3,7 +3,7 @@
 {
     "name": "Date Range",
     "summary": "Manage all kind of date range",
-    "version": "11.0.2.0.1",
+    "version": "11.0.3.0.1",
```
Comment (MiquelRForgeFlow) [inline date_range/__manifest__.py]: the way I see it, this should be:

```suggestion
    "version": "11.0.3.0.0",
```

File: date_range/readme/HISTORY.rst [hunk 8]
```diff
@@ -0,0 +1,23 @@
+11.0.3.0.1 (2019-01-14)
```
Comment (MiquelRForgeFlow) [inline date_range/readme/HISTORY.rst]: And thus:
```suggestion
11.0.3.0.0 (2019-08-08)
```

**Review by MiquelRForgeFlow — COMMENTED** [review]
@pedrobaeza should we proceed to merge this? I can check afterwards if migrations are need if the case.

File: date_range/__manifest__.py (hunk 7 above)
Comment (lmignon) [reply date_range/__manifest__.py]: @mreficent We must use the ocabot merge command to bump the version into the manifest

**Review by lmignon — APPROVED** [review]

File: date_range/__manifest__.py (hunk 7 above)
Comment (MiquelRForgeFlow) [reply date_range/__manifest__.py]: Ok, no problem, but then it should have `11.0.2.0.1` here.

**Comment by ntsirintanis** [comment]
@lmignon @mreficent I believe that this is now ready for merging.

**Comment by MiquelRForgeFlow** [comment]
@ntsirintanis will this be forward-ported?

**Comment by ntsirintanis** [comment]
@mreficent it is in my todo list, but at the moment I can't really work on that.

**Pull request merged**